roles/matrix-bot-maubot/tasks/setup_install.yml

```diff
@@ -0,0 +1,113 @@
+---
+
+- name: Ensure maubot paths exist
+  file:
+    path: "{{ item.path }}"
+    state: directory
+    mode: 0755
+    owner: "{{ matrix_user_username }}"
+    group: "{{ matrix_user_groupname }}"
+  with_items:
+    - {path: "{{ matrix_bot_maubot_base_path }}", when: true}
+    - {path: "{{ matrix_bot_maubot_data_path }}", when: true}
+    - {path: "{{ matrix_bot_maubot_docker_src_files_path }}", when: "{{ matrix_bot_maubot_container_image_self_build }}"}
+  when: "item.when|bool"
+
+- name: Ensure maubot configuration file created
+  template:
+    src: "{{ role_path }}/templates/config/config.yaml.j2"
+    dest: "{{ matrix_bot_maubot_data_path }}/config.yaml"
+    owner: "{{ matrix_user_username }}"
+    group: "{{ matrix_user_groupname }}"
+    mode: "u=rwx"
+
+- name: Generate Maubot proxying configuration for matrix-nginx-proxy
+  set_fact:
+    matrix_bot_maubot_matrix_nginx_proxy_configuration: |
+      location ~ ^/(_matrix/maubot/.*) {
```

Comment:
Mhhh true.
> What's the magic incantation to make the trailing slash optional?
I don't know

Reply:
How about:

```
location /_matrix/maubot/ {
.....
.....
# Just hostname + port here, no paths
set $backend "matrix-maubot:{{ matrix_bot_maubot_port }}";
# Note the trailing slash here
proxy_pass http://$backend/;
```

Reply:
So I tried the following which was inspired by the official docs (did not work, just wanted to leave it here). Will try 
```
- name: Generate Maubot proxying configuration for matrix-nginx-proxy
  set_fact:
    matrix_bot_maubot_matrix_nginx_proxy_configuration: |
      location /_matrix/maubot {
      {% if matrix_nginx_proxy_enabled|default(False) %}
        {# Use the embedded DNS resolver in Docker containers to discover the service #}
         resolver 127.0.0.11 valid=5s;
         proxy_pass http://matrix-bot-maubot:29316";
         proxy_set_header X-Forwarded-For $remote_addr;
      {% else %}
        {# Generic configuration for use outside of our container setup #}
        proxy_pass http://localhost:{{ matrix_bot_maubot_management_interface_port }};
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
      {% endif %}
      }
      location /_matrix/maubot/v1/logs {
      {% if matrix_nginx_proxy_enabled|default(False) %}
        resolver 127.0.0.11 valid=5s;
        proxy_pass http://matrix-bot-maubot:29316";
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
        proxy_set_header X-Forwarded-For $remote_addr;
      {% else %}
        proxy_pass http://localhost:29316;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
        proxy_set_header X-Forwarded-For $remote_addr;
      {% endif %}
      }
```
@spantaleev Your approach redirects `https://synapse.hyteck.de/_matrix/maubot` to `https://synapse.hyteck.de:8443/_matrix/maubot/` and `https://synapse.hyteck.de/_matrix/maubot/` results in a 404

Reply:
I think hardcoding `proxy_pass http://matrix-bot-maubot:29316";` makes nginx try to resolve `matrix-bot-maubot` just once, on startup.

This can fail if the bot service is not running yet. If it works, it's still not good, because if the bot service restarts for whatever reason and gets a new IP address, nginx would still use the old one.

With the `resolver 127.0.0.11 valid=5s;` and `set $backend` dance, we force nginx to resolve the IP address once every 5 seconds.

-----

Why would my approach redirect `https://synapse.hyteck.de/_matrix/maubot` to `https://synapse.hyteck.de:8443/_matrix/maubot/`? Did you mean it forwards it to `https://synapse.hyteck.de:29316/_matrix/maubot/`?

I would expect that it would forward `https://synapse.hyteck.de/_matrix/maubot/SOMETHING` to ``https://synapse.hyteck.de:29316/SOMETHING`. I figured that that is your intention -- removing the `/_matrix/maubot` prefix from the URL when forwarding to the bot. If you'd like **to preserve prefix** instead, then, you'd need to remove the trailing slash from the `proxy_pass` line, like this: `proxy_pass http://$backend;`

You can also try my approach without the trailing slash in the `location` statement, like this: `location /_matrix/maubot`.

Reply:
I honestly don't know why your approach behaves that way. I am puzzled myself.
I also tried `rewrite ^/_matrix_maubot$ ^/_matrix/maubot/ permanent;` and it resulted in the same behavior (rewriting `https://synapse.hyteck.de/_matrix/maubo`t to `https://synapse.hyteck.de:8443/_matrix/maubot/`)

Reply:
Here is a snippet of my nginx conf


```

rewrite ^/_matrix/maubot$ $scheme://$server_name/_matrix/maubot/ permanent;
location ~ ^/(_matrix/maubot/.*) {
     resolver 127.0.0.11 valid=5s;
   set $backend "matrix-bot-maubot:29316/$1";
   proxy_pass http://$backend;
   proxy_set_header Upgrade $http_upgrade;
   proxy_set_header Connection "upgrade";
}

rewrite ^/synapse-admin$ $scheme://$server_name/synapse-admin/ permanent;

```



I included the synapse admin bit because this woks with rewriting to the correct location.


Maybe we can merge this without solving it? I know it is not as satisfying, on the other hand I don't see it as strictly necessary.